library/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -132,80 +227,266 @@
-        msg['update_status'] = idrac.update_mgr.update_from_repo(upd_share,
-                                                                 apply_update,
-                                                                 module.params['reboot'],
-                                                                 module.params['job_wait'])
+        # check if HTTP/HTTPS share
+        if share_name.lower().startswith(('http://', 'https://', 'ftp://')):
+            # Update from http/https/ftp repo is currently supported using
+            # only WS-Man
+            idrac.use_redfish = False
+            result['update_status'] = update_firmware_from_url(idrac,
+                                                               share_name,
+                                                               share_user,
+                                                               share_pwd,
+                                                               catalog_file_name,
+                                                               apply_update,
+                                                               reboot, job_wait,
+                                                               ignore_cert_warning)
+        else:
+            # local mount point is required for SCP based firmware update
+            if idrac.use_redfish and not share_mnt:
+                raise TypeError(msg="Error: \'share_mnt\' is a mandatory argument for firmware update using Server Configuration Profile")
+
+            result['update_status'] = update_firmware_from_net_share(idrac,
+                                                                     share_name,
+                                                                     share_user,
+                                                                     share_pwd,
+                                                                     share_mnt,
+                                                                     catalog_file_name,
+                                                                     apply_update,
+                                                                     reboot,
+                                                                     job_wait)
+
     except RuntimeError as e:
         module.fail_json(msg=str(e))
 
-    if "Status" in msg['update_status']:
-        if msg['update_status']['Status'] == "Success":
+    if "Status" in result['update_status']:
+        if result['update_status']['Status'] == "Success":
+            result['msg'] = 'Successfully created the repository update job.'
             if module.params['job_wait']:
-                msg['changed'] = True
+                remote['msg'] = 'Succesfully completed the repository update job.'
+                result['changed'] = True
         else:
-            module.fail_json(msg='Failed to update firmware.', update_status=msg['update_status'])
-    return msg
+            result['msg'] = 'Failed to update firmware.'
```

Comment:
why are we using single `result` dict for handling both update status and failure message, for any status message there is str variable 'msg' declared in the RETRUN section. Use the same to keep it simple